step-plans/step-plans-core/src/main/java/step/core/artefacts/reports/aggregated/AggregatedReportViewBuilder.java

```diff
@@ -0,0 +1,51 @@
+package step.core.artefacts.reports.aggregated;
+
+import step.core.artefacts.reports.resolvedplan.ResolvedPlanNode;
+import step.core.artefacts.reports.resolvedplan.ResolvedPlanNodeAccessor;
+import step.core.execution.ExecutionEngineContext;
+import step.core.execution.model.Execution;
+import step.core.execution.model.ExecutionAccessor;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+public class AggregatedReportViewBuilder {
+
+    private final String executionId;
+    private final ExecutionAccessor executionAccessor;
+    private final ResolvedPlanNodeAccessor resolvedPlanNodeAccessor;
+    private final ReportNodeTimeSeries reportNodesTimeSeries;
+
+    public AggregatedReportViewBuilder(ExecutionEngineContext executionEngineContext, String executionId) {
+        this.executionId = executionId;
+        this.executionAccessor = executionEngineContext.getExecutionAccessor();
+        this.resolvedPlanNodeAccessor = executionEngineContext.require(ResolvedPlanNodeAccessor.class);
+        reportNodesTimeSeries = executionEngineContext.require(ReportNodeTimeSeries.class);
+    }
+
+    public AggregatedReportView buildAggregatedReportView() {
+        return buildAggregatedReportView(new AggregatedReportViewRequest());
+    }
+
+    public AggregatedReportView buildAggregatedReportView(AggregatedReportViewRequest request) {
+        Objects.requireNonNull(request);
+        Execution execution = executionAccessor.get(executionId);
+        String aggregatedReportRoot = execution.getResolvedPlanRootNodeId();
+        ResolvedPlanNode resolvedPlanNode = resolvedPlanNodeAccessor.get(aggregatedReportRoot);
+        return recursivelyBuildAggregatedReportTree(resolvedPlanNode, request);
+    }
+
+    public static class AggregatedReportViewRequest {
+        ReportNodeTimeSeries.Range range;
```

Comment:
This should probably be public, not `<default>` . I would also make it final for good measure. (Yes, that will require an explicit constructor).